bin/guard

```diff
@@ -2,31 +2,114 @@
 
 require "pathname"
 
-default = Pathname(ENV.fetch("BUNDLE_GEMFILE", "Gemfile"))
-relative_to_binstub = Pathname(__FILE__).realpath + "../../Gemfile"
-gemfile = [default, relative_to_binstub].detect(&:exist?)
-
-if gemfile && !ENV["RUBYGEMS_GEMDEPS"]
-  ENV["BUNDLE_GEMFILE"] = gemfile.to_s
-  require "rubygems"
-  require "bundler/setup"
-else
-  require "rubygems"
-end
+class GuardReloader
+  class Config
+    def using_rubygems?
+      ENV["RUBYGEMS_GEMDEPS"]
+    end
+
+    def setup_rubygems_for_deps
+      require "rubygems"
+    end
+
+    def current_bundler_gemfile
+      ENV["BUNDLE_GEMFILE"]
+    end
+
+    def using_bundler?
```

Comment:
Just to be clear - if `BUNDLE_GEMFILE` is set, it's best to include Bundler to properly deal with the situation (given there are other variables out there, RUBYOPT may need to be set). Without including Bundler here, users would have to rely on Guard properly handling those Bundler/variables edge cases, which makes no sense. By requiring bundler, the number of edge cases are reduced (handled by Bundler).